heartbeat/scheduler/scheduler_test.go

```diff
@@ -240,8 +242,76 @@ func TestScheduler_runRecursiveTask(t *testing.T) {
 	}
 }
 
+func makeTasks(num int, callback func()) TaskFunc {
+	return func(ctx context.Context) []TaskFunc {
+		callback()
+		if num < 1 {
+			return nil
+		}
+		return []TaskFunc{makeTasks(num-1, callback)}
+	}
+}
+
+func TestScheduler_runRecursiveJob(t *testing.T) {
+	tests := []struct {
```

Comment:
Can we add one last test for running jobs when no limit is specified?

Reply:
Done.